Building on Mac osX

I've tried for several hours to build on macOSx Mojave 10.14.5

I have fixed a few problems building Lua, but the build fails with 

```Building md5
env MACOSX_DEPLOYMENT_TARGET=10.3 gcc -bundle -undefined dynamic_lookup -o md5.so lmd5.o -lcrypto
ld: library not found for -lcrypto
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [md5.so] Error 1
```

While ```Building Oniguruma``` is being executed. 

I added ```make MACOSX_DEPLOYMENT_TARGET=10.6 DEFAULT_CC=clang -j8 PREFIX=$BRATPATH/bin/lua``` to line 41 in the build file which helped a few errors.

I currently have ```./configure && CFLAGS="-I/usr/local/Cellar/openssl/1.0.2k/include /usr/local/Cellar//openssl/1.0.2s/lib" make MACOSX_DEPLOYMENT_TARGET=10.6 DEFAULT_CC=clang -j8``` on line 50 with no success. 

Any pointers? I love Brat and would enjoy using it on my Mac, and hopefully contributing more.

--- a/bin/brat
+++ b/bin/brat
@@ -1,7 +1,7 @@
 #!/usr/bin/env lua
 --Set up paths so we can access Brat runtime files
-program_path = debug.getinfo(1, "S").short_src
-program_path = program_path:match("(.+/)bin/brat$")
+program_path = debug.getinfo(1, "S").source
+program_path = program_path:match("([^@].+/)bin/brat$")
 
 if program_path == nil then
   program_path = "../"

--- a/build.sh
+++ b/build.sh
@@ -38,6 +38,8 @@ rm -rf $BRATPATH/bin/lua
 
 #Build Lua
 cd $COMMON/$LUA
+export MACOSX_DEPLOYMENT_TARGET=10.6
+export DEFAULT_CC=clang
 
 make PREFIX=$BRATPATH/bin/lua
 

--- a/src/osx/luafilesystem/Makefile
+++ b/src/osx/luafilesystem/Makefile
@@ -12,7 +12,7 @@ OBJS= src/$T.o
 lib: src/lfs.so
 
 src/lfs.so: $(OBJS)
-	MACOSX_DEPLOYMENT_TARGET="10.3"; export MACOSX_DEPLOYMENT_TARGET; $(CC) $(CFLAGS) $(LIB_OPTION) -o src/lfs.so $(OBJS)
+	$(CC) $(CFLAGS) $(LIB_OPTION) -o src/lfs.so $(OBJS)
 
 install:
 	mkdir -p $(LUA_LIBDIR)

--- a/src/osx/md5/Makefile
+++ b/src/osx/md5/Makefile
@@ -8,11 +8,11 @@ OPENSSL_INC=/usr/local/opt/openssl/include/
 
 # probably no need to change anything below here
 CC= gcc
+INCS= -I$(LUAINC) -I$(OPENSSL_INC)
 CFLAGS= $(INCS) $(DEFS) $(WARN) -O2 -fPIC $G
 WARN= -ansi -pedantic -Wall
-INCS= -I$(LUAINC) -I$(OPENSSL_INC)
 #MAKESO= $(CC) -shared
-MAKESO= env MACOSX_DEPLOYMENT_TARGET=10.3 $(CC) -bundle -undefined dynamic_lookup
+MAKESO= env $(CC) -bundle -undefined dynamic_lookup -L/usr/local/opt/openssl/lib/
 
 # default is md5
 MYNAME=md5

